File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefShortProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import gargoyle.ct.prop.CTNumberProperty;
import org.jetbrains.annotations.NotNull;

public class CTPrefShortProperty extends CTPrefProperty<Short> implements CTNumberProperty<Short> {
    public CTPrefShortProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, (short) 0);
    }

    public CTPrefShortProperty(@NotNull CTPreferencesProvider provider, String name, Short def) {
        super(Short.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefShortProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider
import gargoyle.ct.prop.CTNumberProperty

class CTPrefShortProperty @JvmOverloads constructor(
    provider: CTPreferencesProvider,
    name: String,
    def: Short = 0.toShort()
) : CTPrefProperty<Short>(
    Short::class, provider, name, def
), CTNumberProperty<Short>
